### DISCUSSION TITLE
Is there a programmatic test for whether a function is supported?

### DISCUSSION THREAD

**Discussion by hrbigelow** [KaTeX/KaTeX#3167] [General] [answered]
Hi,

I wondered if there is some programmatic test for whether a function is supported, instead of looking at [https://katex.org/docs/supported.html](https://katex.org/docs/supported.html).

Thanks

**Comment by edemaine** [comment, marked as answer]
I believe the following code will work, e.g. `supported('\\foo')`:
```js
function supported (command) {
  let rendered;
  try {
    rendered = katex.renderToString(command, {throwOnError: false, errorColor: 'INVALID'});
  } catch {
    return true;
  }
  return !/color:INVALID/.test(rendered);
}
```

**Reply by hrbigelow** [reply]
Great, that seems to work, thank you @edemaine
